From 74dd24126a2088c715fc364645beec7a0c040679 Mon Sep 17 00:00:00 2001
From: Guillaume <guillaume.thouvenin@vates.tech>
Date: Mon, 31 Mar 2025 17:06:38 +0200
Subject: [PATCH] [xcp-ng] open/close openflow port

Benjamin tried to upstream a way to open/close ports with SDN controller.
The PR is here https://github.com/xapi-project/xen-api/pull/4188
This patch does it until we have another way to do it.

Signed-off-by: Guillaume <guillaume.thouvenin@vates.tech>
---
 python3/plugins/openvswitch-config-update | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/python3/plugins/openvswitch-config-update b/python3/plugins/openvswitch-config-update
index 834788cf1..9c988db03 100755
--- a/python3/plugins/openvswitch-config-update
+++ b/python3/plugins/openvswitch-config-update
@@ -74,6 +74,7 @@ def update(session, args):
     if not controller and currentControllers:
         delete_cacert()
         runCmd([iptablectl, "close", "6640"])
+        runCmd([iptablectl, "close", "6653"])
         with contextlib.suppress(Exception):
             emergency_reset(session, None)
         removeControllerCfg()
@@ -274,6 +275,7 @@ def setControllerCfg(bootstrap, controller):
                 controller,
             ]
         )
+    runCmd([iptablectl, "open", "6653"])
 
 
 def composeController(protocol, controllerIP, tcpport):
